webapp/lib/charts.dart

```diff
@@ -204,4 +204,27 @@ class DailyTimeseriesLineChartView {
     var chartConfig = new chartjs.ChartConfiguration(type: 'line', data: chartData, options: chartOptions);
     chart = chartjs.Chart(canvas.getContext('2d'), chartConfig);
   }
- }
+}
+
+class SystemMetricsTimeseriesLineChartView extends DailyTimeseriesLineChartView {
+  @override void updateChart(List<Map<DateTime, num>> updatedCountsAtTimestampList, {String timeScaleUnit = 'day', num upperLimit}) {
+    for (var i = 0; i < updatedCountsAtTimestampList.length; i++) {
+      List<chartjs.ChartPoint> timeseriesPoints = [];
+      List<DateTime> sortedDateTimes = updatedCountsAtTimestampList[i].keys.toList()
+        ..sort((t1, t2) => t1.compareTo(t2));
+      for (var datetime in sortedDateTimes) {
+        var value = updatedCountsAtTimestampList[i][datetime];
+        timeseriesPoints.add(
+            new chartjs.ChartPoint(t: datetime.toIso8601String(), y: value));
+      }
+      chartData.datasets[i].data
+        ..clear()
+        ..addAll(timeseriesPoints);
+    }
+    chart.options.scales.xAxes[0].time = (new chartjs.TimeScale(unit: timeScaleUnit));
+    chart.options.scales.yAxes[0].ticks = (new chartjs.LinearTickOptions()
+                                              ..beginAtZero = true
+                                              ..max= upperLimit);
```

Comment:
What happens if upperLimit is `null`? Does it complain or does it just set an appropriate upper limit by itself?

Reply:
I've added a null check and set the parameter to have a null default value